
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Subject: [PATCH] firmware: meson_sm: Add CPU DVFS table index (bin) support
 for Meson-S4

Amlogic Meson-S4 SoCs have a secure monitor firmware call to retrieve
the silicon binning information. Add support for this command which is
going to be used by the cpufreq driver.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 drivers/firmware/meson/meson_sm.c       | 13 +++++++------
 include/linux/firmware/meson/meson_sm.h |  1 +
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/firmware/meson/meson_sm.c b/drivers/firmware/meson/meson_sm.c
index 3ab67aaa9e5da9..4478e5892f3dd7 100644
--- a/drivers/firmware/meson/meson_sm.c
+++ b/drivers/firmware/meson/meson_sm.c
@@ -41,12 +41,13 @@ static const struct meson_sm_chip gxbb_chip = {
 	.cmd_shmem_in_base	= 0x82000020,
 	.cmd_shmem_out_base	= 0x82000021,
 	.cmd = {
-		CMD(SM_EFUSE_READ,	0x82000030),
-		CMD(SM_EFUSE_WRITE,	0x82000031),
-		CMD(SM_EFUSE_USER_MAX,	0x82000033),
-		CMD(SM_GET_CHIP_ID,	0x82000044),
-		CMD(SM_A1_PWRC_SET,	0x82000093),
-		CMD(SM_A1_PWRC_GET,	0x82000095),
+		CMD(SM_EFUSE_READ,		0x82000030),
+		CMD(SM_EFUSE_WRITE,		0x82000031),
+		CMD(SM_EFUSE_USER_MAX,		0x82000033),
+		CMD(SM_GET_CHIP_ID,		0x82000044),
+		CMD(SM_S4_GET_DVFS_TABLE_INDEX,	0x82000088),
+		CMD(SM_A1_PWRC_SET,		0x82000093),
+		CMD(SM_A1_PWRC_GET,		0x82000095),
 		{ /* sentinel */ },
 	},
 };
diff --git a/include/linux/firmware/meson/meson_sm.h b/include/linux/firmware/meson/meson_sm.h
index 8eaf8922ab020a..3b11777d6895f2 100644
--- a/include/linux/firmware/meson/meson_sm.h
+++ b/include/linux/firmware/meson/meson_sm.h
@@ -14,6 +14,7 @@ enum {
 	SM_GET_CHIP_ID,
 	SM_A1_PWRC_SET,
 	SM_A1_PWRC_GET,
+	SM_S4_GET_DVFS_TABLE_INDEX,
 };
 
 struct meson_sm_firmware;
